### ISSUE TITLE
Precompiled binaries are not downloaded on macOS

### ISSUE THREAD

**Issue by toncid** [phusion/passenger#2362]
## Issue report

Fill in as much as possible so that we can understand, find and fix the problem.

**Are you sure this is a bug in Passenger?**

Started happening with v6.0.8 and continued with v6.0.9.

**Please try with the newest version of Passenger to avoid issues that have already been fixed**

Confirmed - using the newest version.

**Question 1: What is the problem?**

Your answer:

The gem is installed normally and on the first run fails to download precompiled binaries, so falls back to building from source. It does not download precompiled binaries, due to some OS version parse error.

The bug seems to be in `PlatformInfo.cxx_binary_compatibility_id`, as it returns the minor macOS version (`11.4`) instead of just the major one (`11`).

I can reproduce the problem after complete gem reinstallation:

```
$ passenger start
 --> Downloading a Phusion Passenger(R) agent binary for your platform
     Could not download https://github.com/phusion/passenger/releases/download/release-6.0.9/agent-x86_64-macosx-11.4.tar.gz: The requested URL returned error: 404 
     Trying next mirror...
     Could not download https://oss-binaries.phusionpassenger.com/binaries/passenger/by_release/6.0.9/agent-x86_64-macosx-11.4.tar.gz: The requested URL returned error: 404 
     Trying next mirror...
     Could not download https://s3.amazonaws.com/phusion-passenger/binaries/passenger/by_release/6.0.9/agent-x86_64-macosx-11.4.tar.gz: The requested URL returned error: 403 Forbidden
     ------------------------------------------
     Sorry, no precompiled agent binary is available for your platform.

---------------------------------------
```

As indicated from the output, the macOS version is correct (v11.4), but since the binary name contains only the major version (`agent-x86_64-macosx-11.tar.gz`), the download fails.

After hardcoding the file name in `download_agent_command.rb`, the agent is downloaded normally, but then the Nginx download fails:

```
$ passenger start
 --> Downloading a Phusion Passenger(R) agent binary for your platform
tar: Failed to set default locale

 --> Installing Nginx 1.20.1 engine
     Could not download https://github.com/phusion/passenger/releases/download/release-6.0.9/nginx-1.20.1-x86_64-macosx-11.4.tar.gz: The requested URL returned error: 404 
     Trying next mirror...
     Could not download https://oss-binaries.phusionpassenger.com/binaries/passenger/by_release/6.0.9/nginx-1.20.1-x86_64-macosx-11.4.tar.gz: The requested URL returned error: 404 
     Trying next mirror...
     Could not download https://s3.amazonaws.com/phusion-passenger/binaries/passenger/by_release/6.0.9/nginx-1.20.1-x86_64-macosx-11.4.tar.gz: The requested URL returned error: 403 Forbidden
     ------------------------------------------
     Sorry, no precompiled Nginx 1.20.1 engine is available for your platform.

---------------------------------------
```

Hardcoding the expected file name in `download_nginx_engine_command.rb` (in this case: `nginx-1.20.1-x86_64-macosx-11.tar.gz`) downloads it and starts the Passenger normally.

**Question 2: Passenger version and integration mode**:

Your answer: Standalone v6.0.9

**Question 3: OS or Linux distro, platform (including version)**:

Your answer: macOS Big Sur v11.4 (x86_64)

**Question 4: Passenger installation method**:

Your answer:
[x] RubyGems + Gemfile
[ ] RubyGems, no Gemfile
[ ] Phusion APT repo
[ ] Phusion YUM repo
[ ] OS X Homebrew
[ ] source tarball
[ ] Other, please specify:

**Question 5: Your app's programming language (including any version managers) and framework (including versions)**:

Your answer: Ruby v2.6.7, RVM, Rails v5.2.6

**Question 6: Are you using a PaaS and/or containerization? If so which one?**

Your answer: Local development environment

**Question 7: Anything else about your setup that we should know?**

Your answer: Nothing particular.

---

We strive for quality and appreciate you taking the time to submit a report! Please note that if you want guaranteed response times and priority issue support we encourage you to join our [enterprise customer](https://www.phusionpassenger.com/features#premium-features) base. They also provide us with the means to continue our high level of open source support!

**Comment by toncid** [comment]
Update: version detection is done by calling `sw_vers`:

```
os_version_string = `sw_vers -productVersion`.strip
```

Since Apple has switched to just using MAJOR as of Big Sur, a simple check like this should be sufficient:

```ruby
os_version =
  if components[0] >= "11"
    components[0]
  else
    "#{components[0]}.#{components[1]}"
  end
```

**Comment by CamJN** [comment]
As you noted, this is due to a bug in the version comparison logic, I've fixed it in 08d9f5b1 and will make a release asap.

**Comment by toncid** [comment]
Thanks!
